### PR TITLE
Update codemeta.json for version 0.4.0 and related release guide instructions

### PR DESCRIPTION
**Pull request by namurphy**
[Codemeta](https://codemeta.github.io/) is an emerging open metadata standard for storing information about software packages.  This PR updates our `codemeta.json` file to be consistent with our version 0.4.0 release, and also adds the author list.  I'm hoping this will (eventually) alleviate some of the tedious aspects of uploading releases to Zenodo.

One continual worry though is that we have some of the same data in different places that will need to be updated simultaneously during releases.  I'm wondering if we could generate the author list that goes in the documentation from `codemeta.json`, or do other similar things.  But, that's for another day (or more likely, for another year).

### CODE REVIEW

**Comment by StanczakDominik** [comment]
So I guess `.zenodo.json` got obsoleted by... what? Does zenodo parse `codemeta.json` now?

File: codemeta.json [hunk 1]
```diff
@@ -1,36 +1,17 @@
...
-  "readme": "https://github.com/PlasmaPy/PlasmaPy/blob/master/README.md",
-  "developmentStatus": "WIP",
-  "buildInstructions": "http://docs.plasmapy.org/en/stable/install.html",
-  "referencePublication": "https://doi.org/10.5281/zenodo.1238132",
-  "funder": [
+    "@context": "https://doi.org/10.5063/schema/codemeta-2.0",
+    "@type": "SoftwareSourceCode",
+    "license": "https://spdx.org/licenses/BSD-2-Clause-Patent",
+    "codeRepository": "https://github.com/PlasmaPy/plasmapy",
+    "contIntegration": "https://dev.azure.com/plasmapy/PlasmaPy/_build/results?buildId=1733&view=results",
```
Comment (StanczakDominik) [inline codemeta.json:6]: Do we want this to point to that particular build (PR 900)?

File: codemeta.json [hunk 2]
```diff
@@ -1,36 +1,17 @@
...
+    "issueTracker": "https://github.com/PlasmaPy/PlasmaPy/issues",
+    "name": "PlasmaPy",
+    "version": "0.4.0",
+    "identifier": "http://doi.org/10.5281/zenodo.4037407",
+    "description": "An open source Python package for plasma research and education",
+    "applicationCategory": "Plasma Physics",
+    "developmentStatus": "wip",
```
Comment (StanczakDominik) [inline codemeta.json:13]: very wip

File: codemeta.json [hunk 3]
```diff
@@ -51,5 +32,446 @@
...
+        },
+        {
+            "@type": "Person",
+            "@id": "https://orcid.org/0000-0001-6291-8843",
+            "givenName": "Dominik",
+            "familyName": "Stańczak",
+            "email": "stanczakdominik@gmail.com",
+            "affiliation": {
+                "@type": "Organization",
+                "name": "University of Warsaw"
```
Comment (StanczakDominik) [inline codemeta.json:96]:
```suggestion
                "name": "IPPLM"
```
let's just say I'm not feeling the affiliation with my university right now. :sweat_smile:

**Review by StanczakDominik — APPROVED** [review]
LGTM, a few minor questions:

File: codemeta.json (hunk 1 above)
Comment (namurphy) [reply codemeta.json:6]: And this, my friend, is why we do code review!

**Comment by namurphy** [comment]
> So I guess `.zenodo.json` got obsoleted by... what? Does zenodo parse `codemeta.json` now?

If I'm remembering correctly, `.zenodo.json` is an experimental feature that never became part of the Zenodo API and never had sufficient documentation anyway.  I don't think I've gotten it to work...though in retrospect maybe it was because the Zenodo uploads included `.zenodo.json` in the zip file, so that `.zenodo.json` wasn't at the top level.  

Cross-reference: #628, https://github.com/zenodo/zenodo/issues/1504, https://github.com/codemeta/codemeta/issues/213

**Pull request merged**